pkg/operators/process/process.go

```diff
@@ -0,0 +1,777 @@
+			enabledFields[field] = true
+		}
+
+		// PID field is always required
+		enabledFields[fieldPID] = true
+
+		gadgetCtx.Logger().Debugf("Enabled fields: %v", fields)
+	}
+
+	// Create a data source for process information
+	ds, err := gadgetCtx.RegisterDataSource(datasource.TypeArray, "processes")
+	if err != nil {
+		return nil, fmt.Errorf("registering processes data source: %w", err)
+	}
+
+	instance := &processOperatorInstance{
+		interval:      interval,
+		enabledFields: enabledFields,
+		done:          make(chan struct{}),
+		dataSource:    ds,
+	}
+
+	// Add fields to the data source based on enabled fields
+	// PID field is always added (it's required)
+	instance.pidField, err = ds.AddField(fieldPID, api.Kind_Int32)
+	if err != nil {
+		return nil, fmt.Errorf("adding pid field: %w", err)
+	}
+
+	// Add optional fields if enabled
+	if instance.enabledFields[fieldPPID] {
+		instance.ppidField, err = ds.AddField(fieldPPID, api.Kind_Int32)
+		if err != nil {
+			return nil, fmt.Errorf("adding ppid field: %w", err)
+		}
+	}
+
+	if instance.enabledFields[fieldCommand] {
+		instance.cmdField, err = ds.AddField(fieldCommand, api.Kind_String)
+		if err != nil {
+			return nil, fmt.Errorf("adding command field: %w", err)
+		}
+	}
+
+	if instance.enabledFields[fieldCPUUsage] {
+		instance.cpuField, err = ds.AddField(fieldCPUUsage, api.Kind_Float64)
+		if err != nil {
+			return nil, fmt.Errorf("adding cpuUsage field: %w", err)
+		}
+	}
+
+	if instance.enabledFields[fieldMemoryRSS] {
+		instance.memoryRSSField, err = ds.AddField(fieldMemoryRSS, api.Kind_Uint64)
+		if err != nil {
+			return nil, fmt.Errorf("adding memoryRSS field: %w", err)
+		}
+	}
+
+	if instance.enabledFields[fieldMemoryVirtual] {
+		instance.memoryVirtualField, err = ds.AddField(fieldMemoryVirtual, api.Kind_Uint64)
+		if err != nil {
+			return nil, fmt.Errorf("adding memoryVirtual field: %w", err)
+		}
+	}
+
+	if instance.enabledFields[fieldThreadCount] {
+		instance.threadCountField, err = ds.AddField(fieldThreadCount, api.Kind_Int32)
+		if err != nil {
+			return nil, fmt.Errorf("adding threadCount field: %w", err)
+		}
+	}
+
+	if instance.enabledFields[fieldState] {
+		instance.stateField, err = ds.AddField(fieldState, api.Kind_String)
+		if err != nil {
+			return nil, fmt.Errorf("adding state field: %w", err)
+		}
+	}
+
+	if instance.enabledFields[fieldUid] {
+		instance.usernameField, err = ds.AddField(fieldUid, api.Kind_Uint32)
+		if err != nil {
+			return nil, fmt.Errorf("adding username field: %w", err)
+		}
+	}
+
+	if instance.enabledFields[fieldStartTime] {
+		instance.startTimeField, err = ds.AddField(fieldStartTime, api.Kind_Uint64)
+		if err != nil {
+			return nil, fmt.Errorf("adding startTime field: %w", err)
+		}
+
+		// Add the formatted start time field
+		instance.startTimeStrField, err = ds.AddField(fieldStartTimeStr, api.Kind_String)
+		if err != nil {
+			return nil, fmt.Errorf("adding startTimeStr field: %w", err)
+		}
+	}
```

Comment:
I'm wondering if this can be automatically handled by the formatters operator. I tried something similar on #4321 but it didn't work, perhaps you can give it a try.

Reply:
I tried it and it kind of worked, but I'm not sure if we shouldn't keep the numeric value as is (e.g. time since boot) and just add the date as usual.